pkgs/servers/search/groonga/default.nix

```diff
@@ -0,0 +1,142 @@
+    "--disable-mruby"
+  ] ++ optionals stdenv.isLinux [
+    "--enable-futex"
+  ];
+
+  outputs = [ "out" "bin" "dev" "lib" "data" "man" "doc" ];
+
+  installFlags = [
+    "sysconfdir=\${data}/etc"
+    "pkgconfigdir=\${dev}/lib/pkgconfig"
+  ];
+
+  postInstall = ''
+    # Make dev propagate libs so that linking works
+    mkdir -p $dev/nix-support
+    echo "$lib" >> $dev/nix-support/propagated-native-build-inputs
+
+    # We expect to be storing nothing in out
+    # rmdir should therefore succeed
+    if ! rmdir $out; then
+      echo "$out unexpected contained files"
+      exit 1
+    fi
+
+    # Out needs to propagate the expected installed files
+    mkdir -p $out/nix-support
+    echo "$bin" >> $out/nix-support/propagated-native-build-inputs
+    echo "$man" >> $out/nix-support/propagated-native-build-inputs
+  '';
+
+  preFixup = ''
+    # Fix the groonga executable to not include paths
+    # It writes the entire config string into the --version option
+    for output in "$out" "$dev" "$man" "$doc"; do
+      sed -i "s,$output,$(echo "$output" | sed 's/./x/g'),g" $bin/bin/groonga
+    done
+    sed -i "/prefix=/d" $bin/sbin/groonga-httpd-restart
+
+    # Fix the groonga.pc having $out references
+    sed -i '/^\(libdir\|includedir\)/!s/^[a-z_]*=.*//g' $dev/lib/pkgconfig/groonga.pc
+  '' + optionalString (!stdenv.isDarwin) ''
+    # Fix the .la file from linking against missing libraries
+    sed \
+      -e "s, -lmsgpack[ '], -L${libmsgpack}/lib\0,g" \
+      -e "s, -llz4[ '], -L${lz4}/lib\0,g" \
+      -e "s, -ljemalloc[ '], -L${jemalloc}/lib\0,g" \
+      -e "s, -lz[ '], -L${zlib}/lib\0,g" \
+      -i $lib/lib/libgroonga.la
+  '';
+
+  postFixup = ''
+    # Make sure there are no extraneous references
+    check_references () {
+      if grep -qr "$1" "$2"; then
+        echo "Extraneous $1 in $2" >&2
+        exit 1
+      fi
+    }
+
+    check_references "$dev" "$out"
+    check_references "$doc" "$out"
+
+    check_references "$out" "$bin"
+    check_references "$dev" "$bin"
+    check_references "$man" "$bin"
+    check_references "$doc" "$bin"
+
+    check_references "$out" "$dev"
+    check_references "$bin" "$dev"
+    check_references "$man" "$dev"
+    check_references "$doc" "$dev"
+
+    check_references "$out" "$lib"
+    check_references "$bin" "$lib"
+    check_references "$dev" "$lib"
+    check_references "$man" "$lib"
+    check_references "$doc" "$lib"
+
+    check_references "$out" "$data"
+    check_references "$bin" "$data"
+    check_references "$dev" "$data"
+    check_references "$lib" "$data"
+    check_references "$man" "$data"
+    check_references "$doc" "$data"
+
+    check_references "$out" "$man"
+    check_references "$bin" "$man"
+    check_references "$dev" "$man"
+    check_references "$lib" "$man"
+    check_references "$data" "$man"
+    check_references "$doc" "$man"
+
+    check_references "$out" "$doc"
+    check_references "$dev" "$doc"
+    check_references "$man" "$doc"
```

Comment:
This is a pretty unscalable way of checking references. I mean, it's basically O(n^2) lines (where n = the number of outputs)... It's also inefficient because it will do lots of I/O that Nix does anyway later.